rmf_site_format/Cargo.toml

```diff
@@ -16,14 +16,17 @@ thiserror = "*"
 glam = { version = "0.24", features = ["serde"] }
 uuid = { version = "1.1", features = ["v4", "serde"] }
 # add features=["bevy"] to a dependent Cargo.toml to get the bevy-related features
-# We depend on a bugfix released specifically in 0.7.3
 bevy = { version = "0.12", optional = true }
 sdformat_rs = { git = "https://github.com/open-rmf/sdf_rust_experimental", rev = "9fc35f2"}
 yaserde = "0.7"
-urdf-rs = "0.7.3"
+# We depend on a bugfix released specifically in 0.7.3
+urdf-rs = { version = "0.7.3", optional = true }
 # Used for lazy initialization of static variable when they are non const
 once_cell = "1"
 pathdiff = "*"
 
 [dev-dependencies]
 float_eq = "1.0"
+
+[features]
+urdf = ["dep:urdf-rs"]
```

Comment:
I think I would recommend making this a default feature. We just need to add
```
default = ["urdf"]
```

Considering the community that we're targeting, I think 
1. It's far more likely that the user will be using URDFs at least somewhat than not using URDFs at all
2. The small proportion of users that will absolutely not use URDF would probably not even notice the dependency
3. Since it's still an optional dependency, they can always disable it if they really don't want it

Reply:
The "idea" behind it was that the site editor itself doesn't support anything `urdf` related at all so I could cut the dependency (in this case with your change I would have probably added a `default_features = off` to its dependency of `rmf_site_format`). But I noticed that we use one [tiny function](https://github.com/open-rmf/rmf_site/blob/983cd3c943e47924c5cf436e30becb3d4cb357c3/rmf_site_editor/src/site_asset_io.rs#L17) to parse `package://` paths and copying the code in doesn't bring any significant gain in dependency count.
TLDR added your suggestion in https://github.com/open-rmf/rmf_site/pull/239/commits/4637e1b6967d6c4d832a8daf6c4d561c9d3ddcdc